src/main/java/com/marklogic/appdeployer/command/AbstractCommand.java

```diff
@@ -151,6 +153,19 @@ protected SaveReceipt saveResource(ResourceManager mgr, CommandContext context,
 	 * @return
 	 */
     protected String adjustPayloadBeforeSavingResource(ResourceManager mgr, CommandContext context, File f, String payload) {
+    	if (context.getAppConfig().getExcludeFields() != null) {
+    		for(String field : context.getAppConfig().getExcludeFields()) {
+    			logger.info("Excluding property %s from payload", field);
+				payload = payloadPropertyFilter.excludeProperty(payload, field);
+			}
+		}
+		if (context.getAppConfig().getIncludeFields() != null) {
```

Comment:
So I was going to say - if both excludeFields and includeFields are set, then an exception should be thrown, which is what ResourceFilenameFilter does. 

I think that behavior makes sense when a single type of resource is being updated. 

But if this is ever used for multiple types of resources, there's some potential for overlap of field names. Perhaps we need to account for this now by requiring a type of resource? e.g. "database:triple-index,servers:url-rewriter". But that would be annoying for the common use case of e.g. "I just want to update the forests on this database, or I want to update everything on this database except the forests". 

So... I think let's assume String[] works here, and if both of these are set, let's throw an exception - i.e. either use mlExcludeProperties or mlIncludeProperties, but not both at the same time.

Reply:
clear